src/mgr/DaemonServer.cc

```diff
@@ -166,55 +166,58 @@ KeyStore *DaemonServer::ms_get_auth1_authorizer_keystore()
   return monc->rotating_secrets.get();
 }
 
-int DaemonServer::ms_handle_authentication(Connection *con)
+void DaemonServer::ms_handle_accept(Connection* con)
 {
-  int ret = 0;
+  std::lock_guard l(lock);
   MgrSession *s = new MgrSession(cct);
-  con->set_priv(s);
-  s->inst.addr = con->get_peer_addr();
+  con->set_priv(RefCountedPtr{s, false});
+  s->inst.addr = con->get_peer_socket_addr();
   s->entity_name = con->peer_name;
   dout(10) << __func__ << " new session " << s << " con " << con
 	   << " entity " << con->peer_name
 	   << " addr " << con->get_peer_addrs()
 	   << dendl;
 
-  AuthCapsInfo &caps_info = con->get_peer_caps_info();
+  const AuthCapsInfo& caps_info = con->get_peer_caps_info();
   if (caps_info.allow_all) {
     dout(10) << " session " << s << " " << s->entity_name
 	     << " allow_all" << dendl;
     s->caps.set_allow_all();
-  }
-
-  if (caps_info.caps.length() > 0) {
```

Comment:
Apparently this line is essential for the mgr. It won't work without it. I haven't updated the commit because allowing a connection to proceed with empty caps is suspicious but I admit to not knowing.